DECLARING_DEPENDENCIES.md

```diff
@@ -0,0 +1,207 @@
+       </dependency>
+     </dependencies>
+  </dependencyManagement>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-storage</artifactId>
+    </dependency>
+  </dependencies>
+```
+
+In this example, since the BOM manages library versions, the
+version of the google-cloud-storage artifact is omitted.
+
+See the "Choosing BOMs" section below if you need to import more than one BOM.
+
+### Gradle
+
+Gradle automatically selects the highest version of any given
+dependency in the dependency tree.
+
+#### Import BOMs to ensure your dependencies are consistent
+
+When a library publishes multiple artifacts, using different versions
+for different artifacts may cause dependency conflicts. To fix this
+problem, you can import a BOM to force consistent versions, as long as
+you are using at least Gradle 4.6. To do this:
+
+- If you are using Gradle 4.x (4.6+):
+  - Add the following to your `settings.gradle`: `enableFeaturePreview('IMPROVED_POM_SUPPORT')`
+- Add a dependency on the BOM for the library you depend on
+- Remove the version from the dependency declarations of the artifacts in that library
+
+For an example, see [gax-java#690](https://github.com/googleapis/gax-java/pull/690/files).
+
+See the "Choosing BOMs" section below if you need to import more than one BOM.
+
+## Choosing BOMs
+
+By default, import the highest-level BOM that your project
+depends on. For example, if you depend on both `grpc-java` and
+`google-cloud-java`, then use `google-cloud-bom` (since `google-cloud-java`
+depends on `grpc-java`). This generally provides you the versions of everything
+you need, because the transitive dependencies of that library should
+all be compatible as well.
+
+However, if your project independently pulls in transitive
+dependencies at different versions, you may also need to specify
+lower-level BOMs to ensure compatibility, since each BOM only controls
+the versions of the library that generates it. Since the GCP Java
+libraries follow semver, you should be able to pick the highest
+version in your dependency tree for any particular library, as long as
+your dependency tree agrees on the same major version. Here is the
+process to follow for each library that has BOM support:
+
+1. Identify the highest version seen for any artifact produced by that library
+  a. For example, if you see `gax:1.34.0` and `gax-grpc:1.42.0`, then use version 1.42.0
+2. Import the BOM corresponding to that library using the highest version you see
+  a. For the example above, use `gax-bom:1.42.0`
+3. Repeat as long as there are other libraries with BOM support where
+   you see multiple versions of the library's artifacts
+
+Note that there will not necessarily be a BOM available at the version
+you need. See the BOM reference section below for the first
+available version of each BOM.
+
+## BOM reference
+
+None of the GCP open source Java libraries were initially released
```

Comment:
I'm not sure this section is needed.

Reply:
I feel pretty confident people will get confused when they can't find BOM artifacts at the version they need. It is marked off as a "reference" section so I don't think it's a huge burden to include (instruction followers can skip it).

Reply:
Also I wanted to note that this is the section where all of the BOM artifact IDs are actually listed; without the table, it would be more work to find them.